src/io.jl

```diff
@@ -0,0 +1,164 @@
+
+"""
+AstroImage(fits::FITS, ext::Int=1)
+
+Given an open FITS file from the FITSIO library,
+load the HDU number `ext` as an AstroImage.
+"""
+AstroImage(fits::FITS, ext::Int=1, args...; kwargs...) = AstroImage(fits[ext], args...; kwargs...)
+
+"""
+AstroImage(hdu::HDU)
+
+Given an open FITS HDU, load it as an AstroImage.
+"""
+AstroImage(hdu::HDU, args...; kwargs...) = AstroImage(read(hdu), args..., read_header(hdu); kwargs...)
+
+"""
+img = AstroImage(filename::AbstractString, ext::Integer=1)
+
+Load an image HDU `ext` from the  FITS file at `filename` as an AstroImage.
+"""
+function AstroImage(filename::AbstractString, ext::Integer=1, args...; kwargs...)
+    return FITS(filename, "r") do fits
+        return AstroImage(fits[ext], args...; kwargs...)
+    end
+end
+"""
+img1, img2 = AstroImage(filename::AbstractString, exts)
+
+Load multiple image HDUs `exts` from an FITS file at `filename` as an AstroImage.
+`exts` must be a tuple, range, :, or array of Integers.
+All listed HDUs in `exts` must be image HDUs or an error will occur.
+
+Example:
+```julia
+img1, img2 = AstroImage("abc.fits", (1,3)) # loads the first and third HDU as images.
+imgs = AstroImage("abc.fits", 1:3) # loads the first three HDUs as images.
+imgs = AstroImage("abc.fits", :) # loads all HDUs as images.
+```
+"""
+function AstroImage(filename::AbstractString, exts::Union{NTuple{N,<:Integer},AbstractArray{<:Integer}}, args...; kwargs...) where {N}
+    return FITS(filename, "r") do fits
+        return map(exts) do ext
+            return AstroImage(fits[ext], args...; kwargs...)
+        end
+    end
+end
+function AstroImage(filename::AbstractString, ::Colon, args...; kwargs...) where {N}
+    return FITS(filename, "r") do fits
+        return map(fits) do hdu
+            return AstroImage(hdu, args...; kwargs...)
+        end
+    end
+end
+
+
+"""
+load(fitsfile::String)
+
+Read and return the data from the first ImageHDU in a FITS file
+as an AstroImage. If no ImageHDUs are present, an error is returned.
+
+load(fitsfile::String, ext::Int)
+
+Read and return the data from the HDU `ext`. If it is an ImageHDU,
+as AstroImage is returned. If it is a TableHDU, a plain Julia
+column table is returned.
+
+load(fitsfile::String, :)
+
+Read and return the data from each HDU in an FITS file. ImageHDUs are
+returned as AstroImage, and TableHDUs are returned as column tables.
+
+load(fitsfile::String, exts::Union{NTuple, AbstractArray})
+
+Read and return the data from the HDUs given by `exts`. ImageHDUs are
+returned as AstroImage, and TableHDUs are returned as column tables.
+
+!! Currently comments on TableHDUs are not supported and are ignored.
```

Comment:
What does `!!` do?  I only know the [`!!!` admonitions](https://juliadocs.github.io/Documenter.jl/stable/showcase/#Admonitions).